plugins/by-name/plantuml-syntax/default.nix

```diff
@@ -1,44 +1,44 @@
 {
   lib,
-  helpers,
-  config,
   pkgs,
   ...
 }:
-with lib;
-{
-  options.plugins.plantuml-syntax = {
-    enable = mkEnableOption "plantuml syntax support";
+let
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.vim-plugin.mkVimPlugin {
+  name = "plantuml-syntax";
+  globalPrefix = "plantuml_";
 
-    package = lib.mkPackageOption pkgs "plantuml-syntax" {
-      default = [
-        "vimPlugins"
-        "plantuml-syntax"
-      ];
-    };
+  maintainers = [ lib.maintainers.GaetanLepage ];
 
-    setMakeprg = mkOption {
-      type = types.bool;
-      default = true;
-      description = "Set the makeprg to 'plantuml'";
-    };
-    executableScript = mkOption {
-      type = types.nullOr types.str;
-      default = null;
-      description = "Set the script to be called with makeprg, default to 'plantuml' in PATH";
+  # TODO introduced 2024-03-02: remove 2024-05-02
+  deprecateExtraConfig = true;
+  optionsRenamedToSettings = [
+    "setMakeprg"
+    "executableScript"
+  ];
+
+  extraOptions = {
+    plantumlPackage = lib.mkPackageOption pkgs "plantuml" {
+      nullable = true;
     };
   };
 
-  config =
-    let
-      cfg = config.plugins.plantuml-syntax;
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
+  extraConfig = cfg: { extraPackages = [ cfg.plantumlPackage ]; };
```

Comment:
I forgot, though... does this need a null check ?

Reply:
Nope, our `extraPackages` option handles `null` values.